Fundamental test coverage of vgrid module reusing existing structure.

jonasbardino opened the pull request:
Fundamental test coverage of `vgrid` module reusing existing structure.

Some tests are disabled on purpose because changes are needed in the tested module. Marked as follow-up.

jonasbardino committed:
Fundamental unit tests for the vgrid module.

jonasbardino committed:
Adjust vgrid self-tests to run as legacy_main for reuse in unit tests. Extended
test coverage to more functions.

jonasbardino committed:
Extend test coverage and add missing docstrings in copy/pasted legacy main
test.

jonasbardino committed:
Cover vgrid_add_owners/members in tests.

albu-diku reviewed (commented):
I think this is a very important area to cover and it will be a boon to have coverage of this in tree.

I've left a comment with suggestions about how I think the tests could be restructured a little for what I think will be a big aid to comprehensibility here: https://github.com/ucphhpc/migrid-sync/pull/371/files#r2472789903. I'm not going to insist hence delivery as a comment rather than requesting changes, but I do feel it worthwhile if possible.

jonasbardino committed:
Address review feedback by mainly splitting up some long joint test cases into
individual 'atomic' tests, while keeping the former as additional 'lifecycle'
tests.

jonasbardino commented on tests/test_mig_shared_vgrid.py:
```diff
@@ -0,0 +1,741 @@
+            self.test_vgrid, 'owners', self.configuration)
+        self.assertEqual(owners, [owner1])
+
+        # Test 2: Prepend new owner
```
Agreed, and I already started splitting similar constructs in other PRs now as you may have recently seen. I have continued here but there's probably room for more. Yet, I think there's no point in delaying this one much further, so I lean towards merging with the latest updates and keeping it in mind for future unit tests or refactoring in a follow-up PR.

jonasbardino replied:
Thanks, and yes while this is a  work in progress I believe we're on the right track. I'll proceed with merging and see if the recent merge of #427 gives rise to obvious test additions as follow-up.

jonasbardino commented:
> I think this is a very important area to cover and it will be a boon to have coverage of this in tree.
> 
> I've left a comment with suggestions about how I think the tests could be restructured a little for what I think will be a big aid to comprehensibility here: https://github.com/ucphhpc/migrid-sync/pull/371/files#r2472789903. I'm not going to insist hence delivery as a comment rather than requesting changes, but I do feel it worthwhile if possible.

Thanks for the review. I've updated and requested another review to also ask for approval before merging.

albu-diku reviewed (approved):
I don't feel quite as confident in terms of having detailed knowledge of the functionality being tested, and as such a word of caution that I can't comment on whether we have exercised all the behaviours we should - but the tests themselves here look good at this point, so approving that aspect.
